tests/modules/s3_datasets_shares/conftest.py

```diff
@@ -253,7 +254,12 @@ def dataset_confidential_fixture(env_fixture, org_fixture, dataset, group) -> S3
 
 @pytest.fixture(scope='module')
 def table_fixture(db, dataset_fixture, table, group, user):
-    table1 = table(dataset=dataset_fixture, name='table1', username=user.username)
+    dataset = dataset_fixture
+    dataset.GlueDatabaseName = dataset_fixture.restricted.GlueDatabaseName
+    dataset.region = dataset_fixture.restricted.region
+    dataset.S3BucketName = dataset_fixture.restricted.S3BucketName
+    dataset.AwsAccountId = dataset_fixture.restricted.AwsAccountId
+    table1 = table(dataset=dataset, name='table1', username=user.username)
```

Comment:
nit: any reason handled this way here and updated `def table` factory for a similar/same required change at `tests/modules/s3_datasets/conftest.py`

Reply:
This is not very elegant, agreed. The issue is that the `dataset_fixture` parameter takes in this case the value of the gql Dataset type. While the table factory accepts db S3Dataset model

Reply:
I am not sure I follow - is it not the exact same dataset and table factories used in `tests/modules/s3_datasets/conftest.py` but in that conftest we define Table factor as

```
@pytest.fixture(scope='module', autouse=True)
def table(db):
    cache = {}

    def factory(dataset: S3Dataset, name, username) -> DatasetTable:
        key = f'{dataset.datasetUri}-{name}'
        if cache.get(key):
            return cache.get(key)
        with db.scoped_session() as session:
            table = DatasetTable(
                name=name,
                label=name,
                owner=username,
                datasetUri=dataset.datasetUri,
                GlueDatabaseName=dataset.restricted.GlueDatabaseName,
                GlueTableName=name,
                region=dataset.restricted.region,
                AWSAccountId=dataset.restricted.AwsAccountId,
                S3BucketName=dataset.restricted.S3BucketName,
                S3Prefix=f'{name}',
            )
...
```

(note the way we set GlueDatabaseName, region, S3BucketName, and S3BucketName)